proposals/partial-events-and-constructors.md

```diff
@@ -0,0 +1,293 @@
+1. Type and ref kind differences between partial declarations which are significant to the runtime result in a compile-time error.
+2. Differences in tuple element names between partial declarations result in a compile-time error.
+3. The declarations must have the same modifiers, though the modifiers may appear in a different order.
+   - Exception: this does not apply to the `extern` modifier which may only appear on the implementing declaration.
+4. All other syntactic differences in the signatures of partial declarations result in a compile-time warning, with the following exceptions:
+   - Attribute lists do not need to match as described [above](#attributes).
+   - Nullable context differences (such as oblivious vs. annotated) do not cause warnings.
+   - Default parameter values do not need to match, but a warning is reported when the implementing constructor declaration has default parameter values
+     (because those would be ignored since only the defining declaration participates in lookup).
+5. A warning occurs when parameter names differ across defining and implementing constructor declarations.
+6. Nullability differences which do not involve oblivious nullability result in warnings.
+
+### Documentation comments
+
+It is permitted to include doc comments on both the defining and implementing declaration.
+Note that doc comments are not supported on event accessors.
+
+When doc comments are present on only one of the declarations of a partial member, those doc comments are used normally
+(surfaced through Roslyn APIs, emitted to the documentation XML file).
+
+When doc comments are present on both declarations of a partial member,
+all the doc comments on the defining declaration are dropped,
+and only the doc comments on the implementing declaration are used.
+
+When parameter names differ between declarations of a partial member,
+`paramref` elements use the parameter names from the declaration associated with the documentation comment in source code.
+For example, a `paramref` on a doc comment placed on an implementing declaration
+refers to the parameter symbols of the implementing declaration using their parameter names.
+This can be confusing, because the metadata signature will use parameter names from the defining declaration.
+It is recommended to ensure that parameter names match across the declarations of a partial member to avoid this confusion.
+
+<!-- ## Drawbacks -->
+
+<!-- Why should we *not* do this? -->
+
+<!-- ## Alternatives -->
+
+<!-- What other designs have been considered? What is the impact of not doing this? -->
+
+## Open questions
+
+<!-- What parts of the design are still undecided? -->
+
+### Member kinds
+
+Do we want partial events, constructors, operators, fields?
+We propose the first two member kinds, but any other subset could be considered.
+(Partial *primary* constructors could be also considered.)
```

Comment:
I find a partial primary constructor syntax difficult to imagine.. maybe simply permitting user to have the same parameter list on multiple partial type declarations?